plugins/modules/xen_orchestra.py

```diff
@@ -0,0 +1,312 @@
+            response = json.loads(self.conn.recv())
+            if 'id' in response and response['id'] == id:
+                return response
+            else:
+                sleep(0.1)
+                waited += 1
+
+        raise AnsibleError(
+            'Method call {method} timed out after {timeout} seconds.'.format(method=method, timeout=self.CALL_TIMEOUT / 10))
+
+    def login(self, user, password):
+        answer = self.call('session.signIn', {
+            'username': user, 'password': password
+        })
+
+        if 'error' in answer:
+            raise self.module.fail_json(
+                'Could not connect: {0}'.format(answer['error']))
+
+        return answer['result']
+
+    def create_vm(self):
+        params = {
+            'template': self.module.params['template'],
+            'name_label': self.module.params['label'],
+            'bootAfterCreate': self.module.params.get('boot_after_create', False)
+        }
+
+        description = self.module.params.get('description')
+        if description:
+            params['name_description'] = description
+
+        answer = self.call('vm.create', params)
+
+        if 'error' in answer:
+            raise self.module.fail_json(
+                'Could not create vm: {0}'.format(answer['error']))
+        
+        return answer['result']
+
+    def restart_vm(self, vm_uid):
+        answer = self.call('vm.restart', {'id': vm_uid, 'force': True })
+
+        if 'error' in answer:
+            raise self.module.fail_json(
+                'Could not restart vm: {0}'.format(answer['error']))
+        
+        return answer['result']
+
+    def stop_vm(self, vm_uid):
+        answer = self.call('vm.stop', {'id': vm_uid, 'force': True})
+
+        if 'error' in answer:
+            # VM is not paused, suspended or running
+            if answer['error']['code'] == VM_STATE_ERROR:
+                return False
+            raise self.module.fail_json(
+                'Could not stop vm: {0}'.format(answer['error']))
+        
+        return answer['result']
+    
+    def start_vm(self, vm_uid):
+        answer = self.call('vm.start', {'id': vm_uid})
+
+        if 'error' in answer:
+            # VM is already started, nothing to do
+            if answer['error']['code'] == VM_STATE_ERROR:
+                return False
+            raise self.module.fail_json(
+                'Could not start vm: {0}'.format(answer['error']))
+        
+        return answer['result']
+
+    def delete_vm(self, vm_uid):
+        answer = self.call('vm.delete', {'id': vm_uid})
+
+        if 'error' in answer:
+            if answer['error']['code'] == OBJECT_NOT_FOUND:
+                return False
+            raise self.module.fail_json(
+                'Could not delete vm: {0}'.format(answer['error']))
+        
+        return answer['result']
+
+def main():
+    if not HAS_WEBSOCKET:
+        raise AnsibleError('This module requires websocket-client 1.0.0 or higher: '
```

Comment:
You cannot use AnsibleError in modules. You must use `modlue.fail_json()`. Also check out https://docs.ansible.com/ansible/devel/dev_guide/testing/sanity/import.html#in-modules how to correctly handle import errors in modules.